include/CXXGraph/Graph/Algorithm/Krager_impl.hpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <random>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
+    std::unordered_map<CXXGraph::id_t, std::unordered_set<CXXGraph::id_t>> adjacency_list;
+
+public:
+    // Other graph methods...
+
+    // Function to add an edge between two nodes
+    void addEdge(const std::string& from_id, const std::string& to_id) {
```

Comment:
this function is a repetition of the already implemented function in `Graph_impl.hpp`, you should use it